lib/chef-dk/policyfile/undo_stack.rb

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'fileutils'
+
+require 'ffi_yajl'
+
+require 'chef-dk/helpers'
+require 'chef-dk/policyfile/undo_record'
+
+module ChefDK
+  module Policyfile
+
+    class UndoStack
+
+      MAX_SIZE = 10
```

Comment:
Sure, that makes sense.  I was thinking about if there was an automated system that removed old policies, but as you say, this tool is mean to be used from a workstation.  :+1: